firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteSerializer.java

```diff
@@ -567,21 +403,19 @@ private FieldMask decodeDocumentMask(DocumentMask mask) {
           (NumericIncrementTransformOperation) transform;
       return DocumentTransform.FieldTransform.newBuilder()
           .setFieldPath(fieldTransform.getFieldPath().canonicalString())
-          .setIncrement(encodeValue(incrementOperation.getOperand()))
+          .setIncrement(incrementOperation.getOperand().getProto())
           .build();
     } else {
       throw fail("Unknown transform: %s", transform);
     }
   }
 
-  private com.google.firestore.v1.ArrayValue encodeArrayTransformElements(
-      List<FieldValue> elements) {
-    com.google.firestore.v1.ArrayValue.Builder arrayBuilder =
-        com.google.firestore.v1.ArrayValue.newBuilder();
-    for (FieldValue subValue : elements) {
-      arrayBuilder.addValues(encodeValue(subValue));
+  private ArrayValue encodeArrayTransformElements(List<FieldValue> elementsProto) {
```

Comment:
`List<FieldValue>` still seems like "elements" more than "elementsProto": we're asking each element to get its underlying proto.